addons/stock/models/stock_package.py

```diff
@@ -312,7 +312,7 @@ def unpack(self):
         self.quant_ids.move_quants(message=self.env._("Quantities unpacked"), unpack=True)
         # Quant clean-up, mostly to avoid multiple quants of the same product. For example, unpack
         # 2 packages of 50, then reserve 100 => a quant of -50 is created at transfer validation.
-        self.quant_ids._quant_tasks()
+        self.quant_ids._unlink_zero_quants()
```

Comment:
you should probably also merge.
P1 LOC A No Pack - 10 units
P1 LOC A Package - 10 units

Unpack
P1 LOC A - 20 units